index.js

```diff
@@ -26,8 +26,8 @@ const takeUntil = notifier => source => (start, sink) => {
         }
         if (t === 2) {
           notifierTalkback = null;
-          done = d;
           if (d != null) {
+            done = d;
```

Comment:
This has fixed an issue because the behavior was different in regards to what gets passed to a sink when:
- inited was false
- inited was true

and both behaviors should be the same.